src/devices/Arduino/ArduinoGpioControllerDriver.cs

```diff
@@ -0,0 +1,303 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Linq;
+using System.Threading;
+
+#pragma warning disable CS1591
+namespace Iot.Device.Arduino
+{
+    internal class ArduinoGpioControllerDriver : GpioDriver
```

Comment:
Also, do you envision only creating these from the ArduinoBoard class exclusively? I ask because most of our drivers are public today, so that people can instantiate them if they need to. I don't have good examples from the top of my head, but would it be possible for people to pluga device (like a LCD screen) and  use pins from both the Arduino board and the phyisical RPi? If so, you would probably want people to be able to create their own driver which would internally use a RPiDriver and ArduinoDriver to forward the calls from pins to the right driver. Does that make sense?

Reply:
First part: Yes, this should only be created from the Board. All devices should only be created from the board, as then the board can handle common stuff. 
Second part: Hmmm... I see your example. But wouldn't it be better to merge the pins at the controller level then and create a MergingGpioController? The bindings take GpioController instances, so doing the mapping on that level seems more natural than abstracting a driver, which is really hardware-bound. This would mean to unseal GpioController (and check what needs to be made virtual).

Reply:
Decision is to unseal the GpioController.